librz/arch/isa/msp430/msp430_il_getset.h

```diff
@@ -0,0 +1,142 @@
+		return MSP430_GETR(operand);
+
+	default: return NULL;
+	}
+}
+
+RZ_OWN RzILOpBitVector *decode_operand(const Msp430Instruction *op, Msp430AddressingMode mode, ut32 operand, ut64 current_addr) {
+	switch (mode) {
+	case MSP430_REG: {
+		RzILOpBitVector *reg = MSP430_GETR(operand);
+		if (word_sizedness(op) & WORD_SIZED_READ) {
+			return reg;
+		}
+		return CAST(8, IL_FALSE, reg);
+	}
+
+	case MSP430_IMM: {
+		if (word_sizedness(op) & WORD_SIZED_READ) {
+			return U16(operand);
+		}
+		return U8(operand);
+	}
+
+	default: {
+		RzILOpBitVector *address = calculate_mem_address(mode, operand, current_addr);
+		if (address != NULL) {
+			if (word_sizedness(op) & WORD_SIZED_READ) {
+				return LOADW(16, address);
+			}
+			return LOADW(8, address);
+		}
+
+		rz_warn_if_reached();
+		return U8(0);
+	}
+	}
+}
+
+RZ_OWN RzILOpBitVector *get_source(const Msp430Instruction *op, ut64 current_addr) {
+	return decode_operand(op, op->src_mode, op->src, current_addr);
+}
+
+RZ_OWN RzILOpBitVector *get_destination(const Msp430Instruction *op, ut64 current_addr) {
+	return decode_operand(op, op->dst_mode, op->dst, current_addr);
+}
+
+typedef struct {
+	RzILOpBitVector *first;
+	RzILOpBitVector *second;
+} BVPair;
+
+// special getter for swpb
+// returns the destination as a "destructured" pair of bytes
+RZ_OWN BVPair get_destination_destructured(const Msp430Instruction *op, ut64 current_addr) {
+	switch (op->dst_mode) {
+	case MSP430_REG: {
+		RzILOpBitVector *reg = MSP430_GETR(op->dst);
+		BVPair res;
+		res.first = CAST(8, IL_FALSE, reg);
+		res.second = CAST(8, IL_FALSE, SHIFTR0(DUP(reg), U8(8)));
+		return res;
+	}
+	default: {
+		RzILOpBitVector *address = calculate_mem_address(op->dst_mode, op->dst, current_addr);
+		if (address != NULL) {
+			BVPair res;
+			res.first = LOAD(address);
+			res.second = LOAD(ADD(address, U16(1)));
+			return res;
+		}
+
+		rz_warn_if_reached();
+		BVPair dummy = { .first = U8(0), .second = U8(0) };
+		return dummy;
+	}
+	}
+}
+
+RZ_OWN RzILOpEffect *set_destination(const Msp430Instruction *op, RzILOpBitVector *new_value, ut64 current_addr) {
+	switch (op->dst_mode) {
+	case MSP430_REG: {
+		if (word_sizedness(op) & WORD_SIZED_WRITE) {
+			return MSP430_SETR(op->dst, new_value);
+		}
+		// the general idea is: First we zero the lower byte through ANDing with 0xFF00
+		//                      Then we assign the lower byte to the (byte-length) result through ORing with 0x00<result>
+		// the overall effect is that only the lower byte is assigned, which is what byte-sized operations do
+		return MSP430_SETR(op->dst, UNSIGNED(16, new_value));
+	}
+
+	default: {
+		RzILOpBitVector *address = calculate_mem_address(op->dst_mode, op->dst, current_addr);
+		if (address != NULL) {
+			if (word_sizedness(op) & WORD_SIZED_WRITE) {
+				return STOREW(address, new_value);
+			}
+			return STORE(address, new_value);
+		}
+
+		rz_warn_if_reached();
+		return NOP();
+	}
+	}
+}
+
+ut8 word_sizedness(const Msp430Instruction *op) {
```

Comment:
```suggestion
static inline ut8 word_sizedness(const Msp430Instruction *op) {
```

Reply:
The ones above should also be `static`, since they are in a header file?

Reply:
@XVilka Forgot what the usual decisions on theses are (since they are quite long). `static` or moving to `.c`?

Reply:
If it's not used across multiple files - just `static` (and `inline` optionally, though often that's reduntant), if it is used across multiple files - a `static inline`, see e.g. `librz/include/rz_vector.h`.

Reply:
In this case the move to a different file is intentional to conceptually seperate main logic from helpers. The helpers are used by every lifter and will clutter and derail the flow of the main `.c` if moved to it, not to mention make it balloon in size to almost double. 

So that's why I seperate them into coherent nice pieces with names like `msp430_il_getset.h` (memory/registers accessors that implements addressing modes) and `msp430_il_flags.h` (flags setters).